lib/modules/onc_program/token_revocation_sequence.rb

```diff
@@ -8,17 +8,57 @@ class TokenRevocationSequence < SequenceBase
 
       test_id_prefix 'TR'
 
-      requires :bulk_client_id, :bulk_jwks_url_auth, :bulk_encryption_method, :bulk_token_endpoint, :bulk_scope
-      defines :bulk_access_token
+      requires :onc_sl_url, :token, :refresh_token, :patient_id, :oauth_token_endpoint 
 
-      test 'Test to be implemented by v1.0' do
+      def encoded_secret(client_id, client_secret)
+        "Basic #{Base64.strict_encode64(client_id + ':' + client_secret)}"
+      end
+
+      test :validate_rejected do
         metadata do
           id '01'
-          link 'https://www.federalregister.gov/documents/2020/05/01/2020-07419/21st-century-cures-act-interoperability-information-blocking-and-the-onc-health-it-certification'
+          name 'Access to Patient resource returns unauthorized after token revocation.'
+          link 'http://www.hl7.org/fhir/smart-app-launch/'
+          description %(
+            This test checks that the Patient resource returns unuathorized after token revocation.
+          )
+        end
+        skip_if @instance.patient_id.nil?, 'Patient ID not provided to test. The patient ID is typically provided during in a SMART launch context.'
+        skip_if @instance.token.nil?, 'Bearer token not provided.  This test verifies that the bearer token can no longer be used to access a Patient resource.'
+
+        @client = FHIR::Client.for_testing_instance(@instance, url_property: 'onc_sl_url')
+        @client.set_bearer_token(@instance.token) unless @client.nil? || @instance.nil? || @instance.token.nil?
+        @client&.monitor_requests
+
+        reply = @client.read(FHIR::Patient, @instance.patient_id)
+
+        assert_response_unauthorized reply
+      end
+
+      test :refresh_rejected do
+        metadata do
+          id '02'
+          name 'Token refresh fails after token revocation.'
+          link 'http://www.hl7.org/fhir/smart-app-launch/'
           description %(
-              Test description
+            This test checks that refreshing token fails after token revokation.
           )
         end
+        skip_if @instance.patient_id.nil?, 'Patient ID not provided to test. The patient ID is typically provided during in a SMART launch context.'
+        skip_if @instance.refresh_token.nil?, 'Patient ID not provided to test. The patient ID is typically provided during in a SMART launch context.'
```

Comment:
Skip message needs to be about refresh tokens.

Reply:
Good point, turns out I don't need patient_id there either.